consensus/core/src/core_thread.rs

```diff
@@ -264,9 +289,21 @@ impl CoreThreadDispatcher for ChannelCoreThreadDispatcher {
             self.highest_received_rounds[block.author()].fetch_max(block.round(), Ordering::AcqRel);
         }
         let (sender, receiver) = oneshot::channel();
-        self.send(CoreThreadCommand::AddBlocks(blocks, sender))
+        self.send(CoreThreadCommand::AddBlocks(blocks.clone(), sender))
             .await;
-        receiver.await.map_err(|e| Shutdown(e.to_string()))
+        let (accepted_block_refs, missing_block_refs) =
+            receiver.await.map_err(|e| Shutdown(e.to_string()))?;
+
+        for block in blocks {
+            let block_ref = block.reference();
+            if accepted_block_refs.contains(&block_ref) {
```

Comment:
I don't think we need to keep track of `highest_accepted_rounds` inside `ChannelCoreThreadDispatcher` and plumb accepted blocks back here. It seems whenever the `highest_accepted_rounds` is needed, it can be read with `dag_state.get_last_cached_block_per_authority(Round::MAX)...`